### PR TITLE
Rewrite client for new public APIs to use `fetch` instead of callables

### PR DESCRIPTION
**Pull request by mqp**
Followup to #227. Again, not changing the contract of the API; just doing all this plumbing work first. After this, now that we are in control, we will start fixing up the actual API contract on the functions and client to be something we endorse for public consumption.

Should be no user-visible changes.

### CODE REVIEW

**Comment by mqp** [comment]
@mantikoros Any thoughts?

File: web/components/answers/answer-bet-panel.tsx [hunk 1]
```diff
@@ -52,22 +52,26 @@ export function AnswerBetPanel(props: {
...
-      setIsSubmitting(false)
-    }
+    })
+      .then((r) => {
+        console.log('placed bet. Result:', r)
+        setIsSubmitting(false)
+        setBetAmount(undefined)
+        props.closePanel()
+      })
+      .catch((e) => {
```
Comment (mantikoros) [inline web/components/answers/answer-bet-panel.tsx:66]: I like the more explicit / less hacky error handling.

**Review by mantikoros — APPROVED** [review]
LGTM. Feel free to merge in if you've made sure everything works.

File: web/components/answers/answer-bet-panel.tsx (hunk 1 above)
Comment (mqp) [reply web/components/answers/answer-bet-panel.tsx:66]: Good! Me too.

**Pull request merged**